File: WorkTestAPI/Services/ProductoService.cs
using WorkTestAPI.Models;
using WorkTestAPI.Repositories;

namespace WorkTestAPI.Services
{
    public class ProductoService : IProductoService
    {
        private readonly IProductoRepository _repo;

        public ProductoService(IProductoRepository repo)
        {
            _repo = repo;
        }

        public async Task<List<Producto>> GetAll()
        {
            return await _repo.GetAll();
        }

        public async Task<Producto?> GetById(int id)
        {
            return await _repo.GetById(id);
        }

        public async Task<Producto> Create(Producto producto)
        {
            return await _repo.Create(producto);
        }

        public async Task<Producto> Update(int id, Producto producto)
        {
            producto.Id = id;
            return await _repo.Update(producto);
        }

        public async Task<bool> Delete(int id)
        {
            return await _repo.Delete(id);
        }
    }
}

File: Frontend/work-test-frontend/src/app/services/producto.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../enviroments/enviroment';

@Injectable({ providedIn: 'root' })
export class ProductoService {

  private api = `${environment.apiUrl}/producto`;

  constructor(private http: HttpClient) {}

  getAll() {
    return this.http.get(this.api);
  }

  create(data: any) {
    return this.http.post(this.api, data);
  }

  update(id: number, data: any) {
    return this.http.put(`${this.api}/${id}`, data);
  }

  delete(id: number) {
    return this.http.delete(`${this.api}/${id}`);
  }

  getById(id: number) {
  return this.http.get(`${environment.apiUrl}/producto/${id}`);
}
}